cli/pcluster/utils.py

```diff
@@ -393,19 +394,41 @@ def verify_stack_creation(stack_name, cfn_client):
         LOGGER.debug(resource_status)
     if status != "CREATE_COMPLETE":
         LOGGER.critical("\nCluster creation failed.  Failed events:")
-        events = get_stack_events(stack_name, raise_on_error=True)
-        for event in events:
-            if event.get("ResourceStatus") == "CREATE_FAILED":
-                LOGGER.info(
-                    "  - %s %s %s",
-                    event.get("ResourceType"),
-                    event.get("LogicalResourceId"),
-                    event.get("ResourceStatusReason"),
-                )
+        _log_stack_failure_recursive(stack_name, indent=2)
```

Comment:
Could you add an example of output in the PR description before and after your patch?